### PR TITLE
ci: enable CD for commits to main

### PR DESCRIPTION
**Pull request by rfratto**
Re-enables continuous delivery for main commits for internal testing.

### CODE REVIEW

File: tools/image-tag-docker [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env sh
+#
+# image-tag-docker runs ./tools/image-tag but sanitizes invalid characters
+# for use in Docker image tags.
+
+TAG=$(sh ./tools/image-tag)
+echo ${TAG//+/-}
```
Comment (rfratto) [inline tools/image-tag-docker:7]: This was being done in a few different places, but also it got complicated to do in the Drone config so I extracted it out to a new script to reduce how many times we have to copy the same sanitization replacement.

File: tools/image-tag [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (rfratto) [inline tools/image-tag:1]: The automation for CD uses `alpine` which doesn't have Bash installed, and it has always used sh. To make sure we continue to support sh I'm explicitly setting it here.

File: .drone/drone.yml [hunk 3]
```diff
@@ -508,6 +508,52 @@ volumes:
...
+  image: alpine
+  name: Create .image-tag
+- image: us.gcr.io/kubernetes-dev/drone/plugins/updater
+  name: Update deployment_tools
+  settings:
+    config_json: |
+      {
+        "git_committer_name": "updater-for-ci[bot]",
+        "git_author_name": "updater-for-ci[bot]",
+        "git_committer_email": "119986603+updater-for-ci[bot]@users.noreply.github.com",
```
Comment (mattdurham) [inline .drone/drone.yml:533]: any danger in exposing this?

Comment (rfratto) [reply .drone/drone.yml:533]: I doubt it, you still need the private key to be able to do anything and other teams have been using the same workflow with the same settings AFAIK (including grafana/agent).

**Review by mattdurham — APPROVED** [review]
lgtm

**Pull request merged**